manifests/charts/istio-control/istio-discovery/files/injection-template.yaml

```diff
@@ -46,7 +46,7 @@ metadata:
     {{- end }}
 {{- if .Values.istio_cni.enabled }}
     {{- if not .Values.istio_cni.chained }}
-    k8s.v1.cni.cncf.io/networks: '{{ appendMultusNetwork (index .ObjectMeta.Annotations `k8s.v1.cni.cncf.io/networks`) `istio-cni` }}',
+    k8s.v1.cni.cncf.io/networks: '{{ appendMultusNetwork (index .ObjectMeta.Annotations `k8s.v1.cni.cncf.io/networks`) `default/istio-cni` }}',
```

Comment:
Actually I'm not sure this will work in all scenarios.

This net-attach-def called `default/istio-cni` will only be exist if `Values.cni.provider == "multus"`: https://github.com/istio/istio/blob/master/manifests/charts/istio-cni/templates/network-attachment-definition.yaml

It is indeed set to `multus` in the openshift profile. If the user is not using the openshift profile (i.e., `Values.cni.provider != "multus"`, this won't work.

Is this an issue at all? Perhaps all cni+multus users are always OpenShift users, then we should not worry?

Reply:
I see a couple of issues:

1. istio_cni.chained is not set and isn't referenced in any istio.io docs. Are all people that are installing cni getting this enabled? So it's partially a documentation bug.
2. We should add a check in the conditional to also check that the provider is multus

Reply:
There are also other K8s users and distributions besides OpenShift that use multus, such as Tanzu. I was even able to get it working in kind (obviously not in production).